doc/sync-label.md

```diff
@@ -0,0 +1,383 @@
+a UI perspective `git-bug` could *preset* remote tracker metadata as if it were
+a `project` assignment. A user can assign a bug to a `project` in UI, where
+under the hood `git-bug` simply stores the associated remote tracker metadata /
+link info. My preference, however, is to keep them as separate entities so that
+`git-bug` created bugs may be assigned to a "project" before a bridge is
+configured for that project.
+
+**open question**: Should it be possible to change a bug's `project` after
+creation? Seems like it would be important to do so, though things might get
+tricky if it has already been exported. For instance if a bug exported over
+`bridge A`, and then the project assignment of that bug is changed to project
+`B`, it might get recreated on the next import from `bridge A` if we're not
+careful.
+
+### private metadata
+
+Any bug which originated outside of `git-bug` (i.e. was `import`ed) is
+permanently `private=false`. Any bug which originates inside of `git-bug` is
+initially created with the `private` field set to a value defined by the
+configuration. If no configuration value is specified then default will be
+`false`. This will match the current behavior meaning bugs created in `git-bug`
+will be exported over and associated with the first bridge on which `export` is
+executed.
+
+**open question**: should the default privacy of a bug be configured only
+globally, or on a per-project basis? If per-project, should a bug's privacy be
+set to that value immediately on assigment to the project? My feeling is that
+making it global is easier to implement and matches my desired usecase better
+but that making it per-project is probably the "right" answer.
+
+## Case Study
+
+Consider the project `git-bug-test` which has a github mirror at
+`cheshirekow.com/git-bug-test`. Ideally we would specify in our git-config
+something like this:
+
+```config
+[git-bug]
+  new-bug-private = true
+[git-bug "bridge.github.github-git-bug-test"]
+	user = cheshirekow
+	project = git-bug-test
+	token = xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
+	import-filter="assignee=cheshirekow"
+```
+
+with the following semantics:
+
+**import**: Only issues matching `assignee=cheshirekow` will be queried from
+github. Any issue that is created is assigned to `project=git-bug-test`. Also
+the issue is marked `private=False (permanent)`.
+
+**export**: Only bugs which have `project=git-bug-test` and `private=False`
+will be considered for export.
+
+
+## User Experience of project assignment
+
+In general, copy the UX for `labels` except that the assignment is unique and
+not multiple. Possibly in some cases it cannot be changed (i.e imported bugs).
+
+### Command line
+
+**open questions**: Should project assignment be basically the same as label
+assignment?
+
+* `git bug ls-project`
+* `git bug project <id> [<project>]`
+
+This begs the question, what happens as more and more metadata are added? Some
```

Comment:
Tricky question again. Some of those will be first-class citizen (= have their own Operation) in git-bug (like the quite obvious assignee). But at the end of the day git-bug can't support every possible workflow or every field existing in other bugtrackers, so choices (or plugin, or fork) will need to be made.

git-bug is designed in a way that bugs are still readable even if an operation type is unknown so it's fine if not everything client support everything, those operations will just be ineffective.

There is also the idea of having a new kind of Operation to store arbitrary fields. @seeduvax is working on that at https://github.com/seeduvax/git-bug/tree/user-defined-attribute

All of that to say that metadata are probably not the right vehicle for those informations. Metadata are more destined to support internal data without history.